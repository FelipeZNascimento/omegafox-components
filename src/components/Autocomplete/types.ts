export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
