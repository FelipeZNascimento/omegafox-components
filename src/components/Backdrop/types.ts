export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
