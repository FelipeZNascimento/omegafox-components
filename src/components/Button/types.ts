export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
