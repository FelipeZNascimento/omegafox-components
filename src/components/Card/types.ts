export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
