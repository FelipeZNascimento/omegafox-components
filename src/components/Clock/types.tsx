import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
