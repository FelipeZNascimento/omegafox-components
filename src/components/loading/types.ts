export interface LoadingProps {
  size?: 'small' | 'regular';
  overlay?: boolean;
  image: string;
}
