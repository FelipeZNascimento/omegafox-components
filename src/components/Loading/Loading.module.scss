@use "../../styling/colors.scss" as colors;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.overlay {
  @include padded(all, none);

  position: absolute;
  max-height: 100vh;
  overflow: hidden;
  z-index: 999;
}

.container {
  @include flex-aligned(center, center);

  flex-direction: column;
  flex: 1;

  &Regular {
    @extend .container;
    @include padded(all, l3);
  }
}

.image {
  @include padded(left, l3);
  @include padded(right, l3);

  width: auto;

  &Shadowed {
    @include color_utils.drop-shadowed;
  }

  &Regular {
    @extend .image;
    max-height: 100px;
  }

  &Small {
    @extend .image;
    max-height: 50px;
  }

  &Headbutt {
    animation: headbutt 3s ease-in infinite;
  }

  &Spin {
    animation: spin 2s ease-in-out infinite;
  }
}

.text {
  @include font-sized(l3);
  @include padded(all, none);
  @include marged(all, none);

  color: colors.$grey2;

  &Shadowed {
    @include color_utils.drop-shadowed;
  }

  &Headbutt {
    animation: headbuttText 3s ease-in-out infinite;
  }
}

@keyframes headbutt {
  20% {
    transform: translate(-25px, 0) rotate(35deg);
  }

  40% {
    transform: translate(-40px, 0) rotate(45deg);
  }

  70% {
    transform: translate(-50px, 0) rotate(50deg);
  }

  80% {
    transform: translate(64px, 0) rotate(60deg);
  }
}

@keyframes headbuttText {
  80% {
    transform: translate(0, 0);
  }
  84% {
    transform: translate(30px, 0);
  }
  90%,
  94% {
    transform: translate(-3px, 0);
  }
  92%,
  96% {
    transform: translate(3px, 0);
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
