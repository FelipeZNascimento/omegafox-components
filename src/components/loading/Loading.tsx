import React from 'react';
import classNames from 'classnames';

import { LoadingProps } from './types';
import styles from './Loading.module.scss';

export const loadingText = 'Carregando...';
export const Loading = ({
  image,
  overlay = false,
  size = 'regular'
}: LoadingProps) => {
  const containerClass = classNames({
    [styles.containerRegular]: size === 'regular',
    [styles.container]: size === 'small',
    [styles.overlay]: overlay
  });

  const imageClass = classNames({
    [styles.imageRegular]: size === 'regular',
    [styles.imageSmall]: size === 'small'
  });

  return (
    <div className={containerClass} data-testid="test-loading">
      <img
        className={imageClass}
        alt="loading spinner"
        src={image}
        data-testid="test-loading__image"
      />
      {size === 'regular' && (
        <p data-testid="test-loading__text" className={styles.text}>
          {loadingText}
        </p>
      )}
    </div>
  );
};
