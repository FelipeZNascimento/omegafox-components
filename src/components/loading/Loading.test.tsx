import React from 'react';
import { render } from '@testing-library/react';
import { Loading, loadingText } from './Loading';
import { LoadingProps } from './types';

const spinnerUrl = './spinner.png';

describe('ButtonComponent', () => {
  const renderComponent = ({
    size,
    overlay,
    image = spinnerUrl
  }: Partial<LoadingProps>) =>
    render(<Loading size={size} overlay={overlay} image={image} />);

  it('should render', () => {
    const { container } = renderComponent({});
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with default values (regular size)', () => {
    const { getByTestId, queryByTestId } = renderComponent({});

    const loading = getByTestId('test-loading');
    const loadingImage = getByTestId('test-loading__image');
    const loadingTextComponent = queryByTestId('test-loading__text');

    expect(loading).toHaveClass('containerRegular');
    expect(loading).not.toHaveClass('overlay');
    expect(loadingImage).toHaveClass('imageRegular');
    expect(loadingImage).toHaveAttribute('src', spinnerUrl);
    expect(loadingTextComponent).toBeVisible();
    expect(loadingTextComponent).toHaveTextContent(loadingText);
  });

  it('should render correctly (small size)', () => {
    const { getByTestId, queryByTestId } = renderComponent({
      size: 'small',
      overlay: true
    });

    const loading = getByTestId('test-loading');
    const loadingImage = getByTestId('test-loading__image');
    const loadingText = queryByTestId('test-loading__text');

    expect(loading).toHaveClass('container');
    expect(loading).toHaveClass('overlay');
    expect(loadingImage).toHaveClass('imageSmall');
    expect(loadingText).toBeNull();
  });
});
