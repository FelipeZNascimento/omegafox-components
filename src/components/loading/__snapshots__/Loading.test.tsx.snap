// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render 1`] = `
<div>
  <div
    class="containerRegular"
    data-testid="test-loading"
  >
    <img
      alt="loading spinner"
      class="imageRegular"
      data-testid="test-loading__image"
      src="./spinner.png"
    />
    <p
      class="text"
      data-testid="test-loading__text"
    >
      Carregando...
    </p>
  </div>
</div>
`;
